src/router/index.ts

```diff
@@ -24,6 +24,14 @@ declare module 'vue-router' {
   }
 }
 
+const setSegmentQuery = async (to: any, from: any, next: any) => {
+  const segmentResetRoutes = ['/shipments', '/purchase-orders', '/returns'];
```

Comment:
This does not handle the case when coming from the settings page. Also doing it in this way will require us to maintain this whenever new page is added in the app.